File: Day_29/Q115.java
import java.util.*;
class Q115{
    static Scanner sc = new Scanner(System.in);

    //Reverse String
    static void reverse_string(String str){
        String rev = "";
        for(int i = str.length() - 1;i >= 0;i--) rev += str.charAt(i);
        System.out.println("Reversed String = " + rev);
    }

    //Palindrome Check
    static void palindrome_check(String str){
        String rev = "";
        for(int i = str.length() - 1;i >= 0;i--) rev += str.charAt(i);
        if(rev.equals(str)) System.out.println("Palindrome String");
        else System.out.println("Not Palindrome String");
    }

    //Count Vowels
    static void count_vowels(String str){
        int count = 0;
        str = str.toLowerCase();
        for(int i = 0;i < str.length();i++){
            char ch = str.charAt(i);
            if(ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u') count++;
        }
        System.out.println("Vowels = " + count);
    }

    //Convert To Uppercase
    static void to_uppercase(String str){
        System.out.println("Uppercase String = " + str.toUpperCase());
    }

    //Convert To Lowercase
    static void to_lowercase(String str){
        System.out.println("Lowercase String = " + str.toLowerCase());
    }

    //Menu
    static void menu(){
        System.out.println("\n1. Reverse String");
        System.out.println("2. Check Palindrome");
        System.out.println("3. Count Vowels");
        System.out.println("4. Convert To Uppercase");
        System.out.println("5. Convert To Lowercase");
        System.out.println("6. Exit");
        System.out.print("Enter your choice: ");
    }

    public static void main(String[] args){
        int choice;
        System.out.print("Enter string: ");
        String str = sc.nextLine();
        do{
            menu();
            choice = sc.nextInt();
            switch(choice){
                case 1: reverse_string(str); break;
                case 2: palindrome_check(str); break;
                case 3: count_vowels(str); break;
                case 4: to_uppercase(str); break;
                case 5: to_lowercase(str); break;
                case 6: System.out.println("Program ended"); break;
                default: System.out.println("Invalid choice");
            }
        }while(choice != 6);
        sc.close();
    }
}

File: Day_29/Q115.cpp
#include <bits/stdc++.h>
using namespace std;

//Reverse String
void reverse_string(string str){
    reverse(str.begin(),str.end());
    cout << "Reversed String = " << str << endl;
}

//Palindrome Check
void palindrome_check(string str){
    string temp = str;
    reverse(temp.begin(),temp.end());
    if(temp == str) cout << "Palindrome String" << endl;
    else cout << "Not Palindrome String" << endl;
}

//Count Vowels
void count_vowels(string str){
    int count = 0;
    for(char ch : str){
        ch = tolower(ch);
        if(ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u') count++;
    }
    cout << "Vowels = " << count << endl;
}

//Convert To Uppercase
void to_uppercase(string str){
    for(int i = 0;i < str.length();i++) str[i] = toupper(str[i]);
    cout << "Uppercase String = " << str << endl;
}

//Convert To Lowercase
void to_lowercase(string str){
    for(int i = 0;i < str.length();i++) str[i] = tolower(str[i]);
    cout << "Lowercase String = " << str << endl;
}

//Menu
void menu(){
    cout << "\n1. Reverse String" << endl;
    cout << "2. Check Palindrome" << endl;
    cout << "3. Count Vowels" << endl;
    cout << "4. Convert To Uppercase" << endl;
    cout << "5. Convert To Lowercase" << endl;
    cout << "6. Exit" << endl;
    cout << "Enter your choice: ";
}

int main(){
    string str;
    int choice;
    cout << "Enter string: ";
    getline(cin,str);
    do{
        menu();
        cin >> choice;
        switch(choice){
            case 1: reverse_string(str); break;
            case 2: palindrome_check(str); break;
            case 3: count_vowels(str); break;
            case 4: to_uppercase(str); break;
            case 5: to_lowercase(str); break;
            case 6: cout << "Program ended" << endl; break;
            default: cout << "Invalid choice" << endl;
        }
    }while(choice != 6);
    return 0;
}
